LCM/dsc/engine/dsc_host/dsc_host.c

```diff
@@ -0,0 +1,324 @@
+#include <stdlib.h>
+#include <locale.h>
+
+#include "dsc_sal.h"
+#include "DSC_Systemcalls.h"
+#include "Resources_LCM.h"
+#include "EngineHelper.h"
+#include "dsc_host.h"
+#include "dsc_library.h"
+#include "lcm/strings.h"
+
+MI_Result GetDSCHostVersion(_In_z_ MI_Char* version, _In_ size_t length)
+{
+    if(Stprintf(version, length, MI_T("Version : %d.%d.%04d"), MajorVersion, MinorVersion, BuildVersion) < 0)
+        return MI_RESULT_FAILED;
+    else
+        return MI_RESULT_OK;
+}
+
+void PrintVersion()
+{
+    MI_Char versionInfo[DSCHOST_VERSION_BUFF_SIZE];
+    GetDSCHostVersion(versionInfo, DSCHOST_VERSION_BUFF_SIZE);
+    Tprintf(versionInfo);
+}
+
+void PrintHelp()
+{
+    Tprintf(MI_T("Usage:\n"));
+    Tprintf(MI_T("dsc_host [--help] [--version]\n"));
+    Tprintf(MI_T("dsc_host <Output Folder Path> <Operation> [Operation Arguments] \n"));
+    Tprintf(MI_T("\n"));
+    Tprintf(MI_T("Supported Operation values are:\n"));
+    Tprintf(MI_T("  SendConfiguration <MOF Document Path>\n"));
+    Tprintf(MI_T("  SendConfigurationApply <MOF Document Path>\n"));
+    Tprintf(MI_T("  GetConfiguration [Configuration Document Path]\n"));
+    Tprintf(MI_T("  ApplyConfiguration\n"));
+    Tprintf(MI_T("  SendMetaConfigurationApply <MOF Document Path> [force]\n"));
+    Tprintf(MI_T("  GetMetaConfiguration\n"));
+    Tprintf(MI_T("  RollBack\n"));
+    Tprintf(MI_T("  TestConfiguration\n"));
+    Tprintf(MI_T("  PerformRequiredConfigurationChecks [flags value]\n"));
+    Tprintf(MI_T("  StopConfiguration [force]\n"));
+    Tprintf(MI_T("  PerformInventory\n"));
+    Tprintf(MI_T("  PerformInventoryOOB [MOF Document Path]\n"));
+    Tprintf(MI_T("\n"));
+    Tprintf(MI_T("Examples:\n"));
+    Tprintf(MI_T("  /opt/dsc/bin/dsc_host /opt/dsc/output SendConfiguration ./new_config.mof\n"));
+    Tprintf(MI_T("  /opt/dsc/bin/dsc_host /opt/dsc/output SendConfigurationApply ./new_config.mof\n"));
+    Tprintf(MI_T("  /opt/dsc/bin/dsc_host /opt/dsc/output GetConfiguration ./GetAuditPolicy.mof\n"));
+    Tprintf(MI_T("  /opt/dsc/bin/dsc_host /opt/dsc/output ApplyConfiguration\n"));
+    Tprintf(MI_T("  /opt/dsc/bin/dsc_host /opt/dsc/output SendMetaConfigurationApply ./new_metaconfig.mof\n"));
+    Tprintf(MI_T("  /opt/dsc/bin/dsc_host /opt/dsc/output GetMetaConfiguration\n"));
+    Tprintf(MI_T("  /opt/dsc/bin/dsc_host /opt/dsc/output RollBack\n"));
+    Tprintf(MI_T("  /opt/dsc/bin/dsc_host /opt/dsc/output TestConfiguration\n"));
+    Tprintf(MI_T("  /opt/dsc/bin/dsc_host /opt/dsc/output PerformRequiredConfigurationChecks 1\n"));
+    Tprintf(MI_T("  /opt/dsc/bin/dsc_host /opt/dsc/output StopConfiguration force\n"));
+    Tprintf(MI_T("  /opt/dsc/bin/dsc_host /opt/dsc/output PerformInventory\n"));
+    Tprintf(MI_T("  /opt/dsc/bin/dsc_host /opt/dsc/output PerformInventoryOOB ./Inventory.mof\n"));
+    Tprintf(MI_T("\n"));
+}
+
+int main(int argc, char *argv[])
+{
+    MI_Instance *extended_error = NULL;
+    MI_Result result = MI_RESULT_OK;
```

Comment:
Is there a plan to remove WMI dependency for full DSC (with one code base) ?   I feel the same code will work in windows (similar to the asm scenario).

Reply:
Not sure what you mean here. DSC v2 does not have a WMI dependency.

---
In reply to: [263591358](https://github.com/Microsoft/PowerShell-DSC-for-Linux/pull/541#discussion_r263591358) [](ancestors = 263591358)

Reply:
I know DSC-V2 does not depend on WMI/OMI.  I am asking if we plan to release 'Full Windows DSC' with no WMI dependency. since that is what you are doing with the OMI here.  ( we can sync offline if this is not clear). 

---
In reply to: [264082023](https://github.com/Microsoft/PowerShell-DSC-for-Linux/pull/541#discussion_r264082023) [](ancestors = 264082023,263591358)